Installing the `debugpy-python2` gadget fails

### Description

Installing `debugpy-python2` fails on both Linux and Windows.


### Is it reproducible in vim?

Reproducible in Vim

### Works with clean config?

Yes

### Sample project works?

Reproducible with sample project

### Minimal Reproduction

List of steps to reproduce:

<details>
  <summary>
$ ~/.vim/plugged/vimspector/install_gadget.py --force-enable-python2
  </summary>
<pre>
Installing debugpy-python2@1.5.1...
Checksum matches for ~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/v1.5.1.zip, using it
Extracting ~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/v1.5.1.zip to ~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/root
Removing dir ~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/root
~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/root/debugpy-1.5.1/versioneer.py:564: SyntaxWarning: invalid escape sequence '\s'
  mo = re.search(r'=\s*"(.*)"', line)
Compiling extension modules (set SKIP_CYTHON_BUILD=1 to omit)
running build_ext
building '_pydevd_bundle.pydevd_cython' extension
creating build/temp.linux-x86_64-cpython-313/_pydevd_bundle
x86_64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -DPy_BUILD_CORE_MODULE=1 -I/usr/include/python3.13 -c _pydevd_bundle/pydevd_cython.c -o build/temp.linux-x86_64-cpython-313/_pydevd_bundle/pydevd_cython.o -flto -O3
_pydevd_bundle/pydevd_cython.c:216:12: fatal error: longintrepr.h: Nu există un astfel de fișier sau director
  216 |   #include "longintrepr.h"
      |            ^~~~~~~~~~~~~~~
compilation terminated.
error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
Traceback (most recent call last):
  File "~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/root/debugpy-1.5.1/setup.py", line 150, in <module>
    version=versioneer.get_version(),
            ~~~~~~~~~~~~~~~~~~~~~~^^
  File "~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/root/debugpy-1.5.1/versioneer.py", line 1481, in get_version
    return get_versions()["version"]
           ~~~~~~~~~~~~^^
  File "~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/root/debugpy-1.5.1/versioneer.py", line 1413, in get_versions
    cfg = get_config_from_root(root)
  File "~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/root/debugpy-1.5.1/versioneer.py", line 342, in get_config_from_root
    parser = configparser.SafeConfigParser()
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
Traceback (most recent call last):
  File "~/.vim/plugged/vimspector/python3/vimspector/installer.py", line 579, in InstallGadget
    gadget[ 'do' ]( name, root, spec )
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "~/.vim/plugged/vimspector/python3/vimspector/gadgets.py", line 150, in <lambda>
    'do': lambda name, root, gadget: installer.InstallDebugpy( name,
                                     ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^
                                                               root,
                                                               ^^^^^
                                                               gadget ),
                                                               ^^^^^^^^
  File "~/.vim/plugged/vimspector/python3/vimspector/installer.py", line 407, in InstallDebugpy
    CheckCall( [ sys.executable,
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^
                 'setup.py',
                 ^^^^^^^^^^^
                 'build',
                 ^^^^^^^^
                 '--build-platlib', os.path.join( 'build', 'lib' ) ] )
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.vim/plugged/vimspector/python3/vimspector/installer.py", line 109, in CheckCall
    subprocess.check_call( cmd, *args, **kwargs )
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.13/subprocess.py", line 419, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/usr/bin/python3', 'setup.py', 'build', '--build-platlib', 'build/lib']' returned non-zero exit status 1.
 - FAILED installing debugpy-python2: Command '['/usr/bin/python3', 'setup.py', 'build', '--build-platlib', 'build/lib']' returned non-zero exit status 1.
Failed to install adapters:
 * debugpy-python2
</pre>
 </details>

<details>
  <summary>
:VimspectorInstall debugpy-python2
  </summary>
<pre>
Installing debugpy-python2@1.5.1...
~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/root/debugpy-1.5.1/versioneer.py:564: SyntaxWarning: invalid escape sequence '\s'
  mo = re.search(r'=\s*"(.*)"', line)
running build_ext
building '_pydevd_bundle.pydevd_cython' extension
creating build/temp.linux-x86_64-cpython-313/_pydevd_bundle
x86_64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -DPy_BUILD_CORE_MODULE=1 -I/usr/include/python3.13 -c _pydevd_bundle/pydevd_cython.c -o build/temp.linux-x86_64-cpython-313/_pydevd_bundle/pydevd_cython.o -flto -O3
_pydevd_bundle/pydevd_cython.c:216:12: fatal error: longintrepr.h: No such file or directory
  216 |   #include "longintrepr.h"
      |            ^~~~~~~~~~~~~~~
compilation terminated.
error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
Compiling extension modules (set SKIP_CYTHON_BUILD=1 to omit)
Traceback (most recent call last):
  File "\~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/root/debugpy-1.5.1/setup.py", line 150, in <module>
    version=versioneer.get_version(),
            ~~~~~~~~~~~~~~~~~~~~~~^^
  File "\~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/root/debugpy-1.5.1/versioneer.py", line 1481, in get_version
    return get_versions()["version"]
           ~~~~~~~~~~~~^^
  File "~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/root/debugpy-1.5.1/versioneer.py", line 1413, in get_versions
    cfg = get_config_from_root(root)
  File "~/.vim/plugged/vimspector/gadgets/linux/download/debugpy-python2/1.5.1/root/debugpy-1.5.1/versioneer.py", line 342, in get_config_from_root
    parser = configparser.SafeConfigParser()
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?

 - FAILED installing debugpy-python2: Command '['/usr/bin/python3', 'setup.py', 'build', '--build-platlib', 'build/lib']' returned non-zero exit status 1.

***NOTE***: You set --basedir to \~/.vim/plugged/vimspector/python3/vimspector/../... Therefore you _must_ ensure this is in your vimrc:
let g:vimspector_base_dir='\~/.vim/plugged/vimspector'
Failed to install adapters:
 * debugpy-python2
Re-run with --verbose for more info on failures
</pre>
</details>

The issue is that `setup.py build --build-platlib build/lib` for debugpy-python2 is called with python3 instead of python2.

Reproducible on Ubuntu 25.04 and Windows 7. I have not tested any other OS.


### Expected Behaviour

Assuming that `python2` is available in $PATH, gadget installation should succeed.

### Actual Behaviour

Installation fails.

### Additional information

_No response_

### Installation Type

Plugin Manager (git clone)

### Vimspector version

a1ee02d4e5df05d411e93740465db1177ba42b4c

### Debug Info

```Text

```

### Log file

```Text

```

### Vim version

```Text
VIM - Vi IMproved 9.1 (2024 Jan 02, compiled Apr 03 2025 14:08:49)
Included patches: 1-948, 950-967
```

### Python version

_No response_

### Neovim diagnostics

```Text

```

### Operating System

_No response_

### Declaration

- [x] I have read and understood [CONTRIBUTING.md](https://github.com/puremourning/vimspector/blob/master/CONTRIBUTING.md#diagnostics)
- [x] I have read and understood the [CODE_OF_CONDUCT.md](https://github.com/puremourning/vimspector/blob/master/CODE_OF_CONDUCT.md)

--- a/README.md
+++ b/README.md
@@ -189,7 +189,6 @@ runtime dependencies). They are categorised by their level of support:
 | PHP                     | Experimental          | `--force-enable-php`                | vscode-php-debug                     | Node, PHP, XDEBUG                            |
 | C# (dotnet core)        | Tested                | `--force-enable-csharp`             | netcoredbg                           | DotNet core                                  |
 | F#, VB, etc.            | Supported             | `--force-enable-[fsharp,vbnet]`     | netcoredbg                           | DotNet core                                  |
-| Python 2                | Legacy [Unmaintained] | `--force-enable-python2`            | debugpy-python2                      | Python 2.7                                   |
 
 ## Other languages
 
@@ -1891,9 +1890,6 @@ Example:
 * Install with `install_gadget.py --enable-python` or `:VimspectorInstall
   debugpy`, ideally requires a working compiler and the python development
   headers/libs to build a C python extension for performance.
-* ***NOTE***: Debugpy no longer supports python 2. In order to continue to debug
-  python 2 applications, use the `debugpy-python2` adapter after installing the
-  `debugpy-python2` gadget.
 
 * Full options: https://github.com/microsoft/debugpy/wiki/Debug-configuration-settings
 
@@ -1967,24 +1963,8 @@ an example of getting Vimspector to remotely launch and attach.
 
 ### Python 2
 
-In order to continue to debug python 2 applications, ensure that you install the
-`debugpy-python2` gadget (e.g. `--force-enable-python2` or
-`:VimspectorInstall debugpy-python2`), and then change your configuration to
-use:
-
-```json
-{
-  "configurations": {
-    "Python Attach": {
-      "adapter": "debugpy-python2",
-      // ...
-    }
-  }
-}
-
-```
-
-for example
+Python 2 is no longer supported, having been out of support by both Python and
+debugpy for years.
 
 ## TCL
 

--- a/python3/vimspector/gadgets.py
+++ b/python3/vimspector/gadgets.py
@@ -156,36 +156,6 @@ GADGETS = {
       }
     },
   },
-  'debugpy-python2': {
-    'language': 'python2',
-    'enabled': False,
-    'download': {
-      'url': 'https://github.com/microsoft/debugpy/archive/${file_name}'
-    },
-    'all': {
-      # Don't update - this is the last version that supports python2
-      'version': '1.5.1',
-      'file_name': 'v1.5.1.zip',
-      'checksum':
-        '00cf8235b88880bc2d8f59e8f6585208a43e6f14017cdf11d3a0bb2aeb4fff79'
-    },
-    'do': lambda name, root, gadget: installer.InstallDebugpy( name,
-                                                               root,
-                                                               gadget ),
-    'adapters': {
-      'debugpy-python2': {
-        "command": [
-          sys.executable,
-          "${gadgetDir}/debugpy-python2/build/lib/debugpy/adapter"
-        ],
-        "name": "debugpy",
-        "configuration": {
-          "python": sys.executable
-        },
-        'custom_handler': 'vimspector.custom.python.Debugpy'
-      }
-    },
-  },
   'vscode-java-debug': {
     'language': 'java',
     'enabled': False,

